apps/studio/src/features/mail/service.ts

```diff
@@ -0,0 +1,40 @@
+import type { InvitationEmailTemplateData } from "./templates"
+import { createBaseLogger } from "~/lib/logger"
+import { isValidEmail } from "~/utils/email"
+import { sendMail } from "../../lib/mail"
+import { templates } from "./templates/templates"
+
+const logger = createBaseLogger({ path: "lib/mail/service" })
+
+export async function sendInvitation(
+  email: string,
+  data: InvitationEmailTemplateData,
+): Promise<void> {
+  if (!isValidEmail(email)) {
+    logger.error({
+      error: "Invalid email format",
+      email,
+    })
+    throw new Error("Invalid email format")
+  }
+
+  const template = templates.invitation({
```

Comment:
I really like how you've done this template grouping! it makes alot of sense - just wanted to ask, is there some base requirement that templates adhere to? don't have to do anything yet but we might benefit from making `templates` a mapping of that type so we know all templates share some base guarantees

Reply:
makes sense, added some form of type requirements for `templates` to ensure it strictly follows a base template in https://github.com/opengovsg/isomer/pull/1123/commits/b1296ebe250ee9b544dc4854735b71c14dc23fe1